inputs/demand/industry/industry_useful_demand_for_chemical_refineries.ad

```diff
@@ -1,17 +1,148 @@
+# This input statement updates the useful refinery output and energetic demand.
+# It has two operation modes:
+# - With refinery output in dataset start year: a relative change is calculated and
+#   the energetic and non energetic useful demand are updated accordingly.
+# - Without refinery output in dataset start year: default values
+#   of a 'typical refinery' are taken assumed and updated according to the set refinery output.
+#
+# For more information on the 'typical refinery' conversion values (based on EU27_european_union_27_countries
+# dataset), see etdataset/inputs_source_analyses/demand/industry.
+
 - query =
-    EACH(
-    UPDATE_WITH_FACTOR(V(industry_useful_demand_for_chemical_refineries_coal_non_energetic), preset_demand, USER_INPUT()),
-    UPDATE_WITH_FACTOR(V(industry_useful_demand_for_chemical_refineries_crude_oil_non_energetic), preset_demand, USER_INPUT()),
-    UPDATE_WITH_FACTOR(V(industry_useful_demand_for_chemical_refineries_electricity), preset_demand, USER_INPUT()),
-    UPDATE_WITH_FACTOR(V(industry_useful_demand_for_chemical_refineries_network_gas_non_energetic), preset_demand, USER_INPUT()),
-    UPDATE_WITH_FACTOR(V(industry_useful_demand_for_chemical_refineries_useable_heat), preset_demand, USER_INPUT()),
-    UPDATE_WITH_FACTOR(V(industry_useful_demand_for_chemical_refineries_wood_pellets_non_energetic), preset_demand, USER_INPUT()))
-- priority = 0
-- max_value = 300.0
+    present_value = QUERY_PRESENT(-> {Q(industry_useful_output_industry_refinery_transofrmation_crude_oil)});
+    relative_value = USER_INPUT() / present_value;
+
+    electricity_input_conversion_typical_refinery      = 0.00514;
+    heat_input_conversion_typical_refinery             = 0.05341;
```

Comment:
I don't think these variables are clearly defined: "input conversion" sounds to me like some kind of electricity and heat is going into the transformation node. Also, I would clearly distinguish between useable_heat and steam_hot_water. The generic "heat" may be confusing.